orchestration/dagster_orchestration/hca_manage/diff_dirs.py

```diff
@@ -0,0 +1,41 @@
+"""
+Compares the contents of two google storage paths via their md5
+"""
+import argparse
+import logging
+
+import google.auth
+from google.cloud import storage
+
+logging.basicConfig(level=logging.INFO)
+
+
+def run(project, source_bucket, source_prefix, target_bucket, target_prefix):
+    creds, _ = google.auth.default()
+    storage_client = storage.Client(project=project, credentials=creds)
+    expected_blobs = {blob.name.replace(source_prefix, ''): blob.md5_hash
+                      for blob in storage_client.list_blobs(source_bucket,
+                                                            prefix=source_prefix)}
+
+    logging.info(f"project = {project}, "
+                 f"source_bucket = {source_bucket}, "
+                 f"source_prefix = {source_prefix}, "
+                 f"target_bucket = {target_bucket}, "
+                 f"target_prefix = {target_prefix}")
+    output_blobs = {blob.name.replace(target_prefix, ''): blob.md5_hash
+                    for blob in storage_client.list_blobs(target_bucket,
+                                                          prefix=target_prefix)}
+
+    assert expected_blobs == output_blobs, "Output results differ from expected"
```

Comment:
does this get logged as an error?

Reply:
Yes; it will raise an AssertionError and the script will return 1 like so:

```
poetry run python diff_dirs.py -sb broad-dsp-monster-hca-dev-test-storage -sp integration/ebi_small/expected -tb foo -tp bar
INFO:root:project = None, source_bucket = broad-dsp-monster-hca-dev-test-storage, source_prefix = integration/ebi_small/expected, target_bucket = broad-dsp-monster-hca-dev-temp-storage, target_prefix = integration/output_
Traceback (most recent call last):
  File "/Users/aherbst/src/hca-ingest/orchestration/dagster_orchestration/hca_manage/diff_dirs.py", line 41, in <module>
    run(args.project, args.source_bucket, args.source_prefix, args.target_bucket, args.target_prefix)
  File "/Users/aherbst/src/hca-ingest/orchestration/dagster_orchestration/hca_manage/diff_dirs.py", line 29, in run
    assert expected_blobs == output_blobs, "Output results differ from expected"
AssertionError: Output results differ from expected

-- 

» echo $?                     
1
```